app/views/vm_common/_reconfigure.html.haml

```diff
@@ -343,10 +343,18 @@
                                                               "ng-click" => "vm.cancelAddRemoveNetworkAdapter(networkAdapter)"}= _('Cancel Add')
         %tr{"ng-if"       => "vm.reconfigureModel.showDropDownNetwork",
             "ng-form"     => "rowForm"}
-          %td
+          %td{"ng-if" => "!vm.isVmwareCloud()"}
+          %td{"ng-if" => "vm.isVmwareCloud()"}
+            = select_tag('adapterIdx',
+                       options_for_select([["<#{_('Choose')}>", '']] + @avail_adapter_names, :disabled => ["<#{_('Choose')}>", nil]),
+                       "ng-model"                    => "vm.reconfigureModel.name",
+                       "ng-change"                   => "",
+                       "data-width"                  => "auto",
+                       "required"                    => "true",
+                       "selectpicker-for-select-tag" => "")
           %td
             = select_tag('vLan',
-                       options_for_select([["<#{_('Choose')}>", '']] + @vlan_options, :disabled => ["<#{_('Choose')}>", nil]),
+                       options_for_select([["<#{_('Choose')}>", '']] + @vlan_options),
```

Comment:
Is chosing no lans is suddenly ok regardless of type?

Reply:
No - we make `<Choose>` option selectable in drop-down, but then disable `Confirm Add` button if choosing no lans isn't feasible behavior for provider type (see L367).

Reply:
Aah, thanks, makes sense :)

Reply:
Actually.. no .. the field is still marked as required

Reply:
As far as I've tested adding `"required" => "true"` or `"required" => "false"` makes no difference at all (is ignored for some reason) but no problem, I'll add that.

Reply:
Thanks, yeah, the form does not actually use that information, but it really should :)